lib/linked_list.rb

```diff
@@ -62,29 +64,43 @@ def length
     # Time Complexity:  
     # Space Complexity
     def get_at_index(index)
-      raise NotImplementedError
+      current_node = @head
+      while current_node.next?
+        current_node = current_node.next
     end
 
     # method to print all the values in the linked list
     # Time Complexity:  
     # Space Complexity
     def visit
-      raise NotImplementedError
+      
     end
 
-    # method to delete the first node found with specified value
-    # Time Complexity:  
-    # Space Complexity
+    #### method to delete the first node found with specified value
+    # Time Complexity:  o(n)
+    # Space Complexity: o(n)
     def delete(value)
-      raise NotImplementedError
+      if self.value = true
+        remove self
     end
 
-    # method to reverse the singly linked list
+    #### method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
     # Time Complexity:  
-    # Space Complexity
+    # Space Complexity:
+    # do with +  w/o recursion ?
     def reverse

```

Comment:
👍 

Since you didn't make any external data structures, your space complexity here is O(1), and since you have to traverse the list once, your time complexity is O(n).  You can usually guess this by the fact that there is one and only one loop.